src/vt_stability.jl

```diff
@@ -104,21 +104,18 @@ function vt_stability_optim_try!(
     Dfunc!::Function,  # функция D вида D!(η', ∇D_) -> D::Number
     maxstep::Function,
 )
-    is_converged = false
-
-    try
-        result = Downhill.optimize!(Dfunc!, optmethod, η_;
-            gtol=1e-3,
-            maxiter=1000,
-            constrain_step=maxstep,
-            reset=false,
-        )
-        is_converged = result.converged
-    catch e
-        @warn e
-        is_converged = false
+    optresult = Downhill.optimize!(Dfunc!, optmethod, η_;
+        gtol=1e-3,
+        maxiter=1000,
+        constrain_step=maxstep,
+        reset=false,
+    )
+
+    if optresult.converged
+        D_min, _ = Dfunc!(optmethod.x, grad_)
+    else
+        D_min = NaN
     end
```

Comment:
No `try/catch` block at all. If Downhill has an error, it explicitly propagates. Also, if no convergence found, `D_min` is set to `NaN`.